### PR TITLE
Implement editing and encoding

### PR DESCRIPTION
**Pull request by GNUp**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -153,20 +134,41 @@ interface ItemState {
     string: string | null;
     number: string | null;
     timestamp: string | null;
+    [key: string]: string | string[] | null;
   };
+  value: Buffer | Buffer[];
```
Comment (foriequal0) [inline src/App.tsx]: Can this type can express a nested list such as `[1,[2,[3,[4]]]` ?

Comment (GNUp) [reply src/App.tsx]: I changed it in 8c88210

**Comment by GNUp** [comment]
For your convenience,

**Comment by foriequal0** [comment]
Can you fix bugs first?

- [ ] Bug 1
  1. Input "0"
  2. See "hex/00"
  3. Click "List" button
  4. "hex/00" is changed to a List and its child is "hex/0x80" (I think this shouldn be)
  5. Click "Encode" and see encoded text is "80" (rlp(["0x80"]) isn't definately not "80")

- [ ] Bug 2
  Trying to encode "number/1234" is rlp encode error.

- [ ] Feature change request?
  1. Input "8200"
  2. See "hex/00".
  3. Cannot change the type to the string when I want to the type to the string.

- [ ] Unintuitive UI
  "+" button on the right of a "List" label inserts new item in front of the list, however "-" button removes entire list. "+" button on the right of the item inserts an item after that, however there is no equivalent button for a list.
I think this is not intuitive. 
A button to insert item in front of the list should be in front of the list, and The button "+" on the right of the list should append an item after the list.

**Comment by GNUp** [comment]
@foriequal0 
1. fix the bug in type changing
2. change the UI more intuitive

**Pull request merged**